examples/even-odd-list-with-type-level-booleans

```diff
@@ -0,0 +1,45 @@
+/// A list of type `a` that uses a Boolean at the type level to track
+/// whether its length is even or odd.
+enum EOList[a: Type, even: Bool] {
+    case Nil
+    case Cons(a, EOList[a, not even])
+}
+
+/// Returns an empty list of type `a` with even length (which is 0).
+def mkNil(): EOList[a, true] = EOList.Nil
+
+/// Attaches an element to the front of the list.
+def cons(x: a, l: EOList[a, e]): EOList[a, not e] =
+    EOList.Cons(x, l)
+
+/// Returns the head of the list.
+def head(l: EOList[a, e]): a = match l {
+    case EOList.Cons(x, _) => x
+    case EOList.Nil        => bug!("Empty input not allowed")
+}
+
+/// Returns the tail of the list.
```

Comment:
/// Returns the tail of the list. Note how this changes the evenness of the list.